project/src/main/java/com/google/sps/servlets/CandidateServlet.java

```diff
@@ -64,24 +63,70 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
   // @TODO [Function for getting (1) official election/candidate information from the database.]
+  private Candidate getCandidateData(String candidateId, String electionName) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query candidateQuery =
+        new Query("Candidate")
+            .setFilter(
+                new FilterPredicate(
+                    "__key__",
+                    FilterOperator.EQUAL,
+                    KeyFactory.createKey("Candidate", Long.parseLong(candidateId))));
+    PreparedQuery candidateQueryResult = datastore.prepare(candidateQuery);
+    Entity candidateData = candidateQueryResult.asSingleEntity();
+
+    Query electionQuery =
+        new Query("Election")
+            .setFilter(
+                new FilterPredicate(
+                    "__key__",
+                    FilterOperator.EQUAL,
+                    KeyFactory.createKey("Election", electionName)));
+    PreparedQuery electionQueryResult = datastore.prepare(electionQuery);
+    Entity election = electionQueryResult.asSingleEntity();
+    List<String> electionCandidateIdList = (List<String>) election.getProperty("candidateIds");
+    List<String> electionPositionList = (List<String>) election.getProperty("candidatePositions");
+    List<Boolean> electionIsIncumbentList =
+        (List<Boolean>) election.getProperty("candidateIncumbency");
+    int candidateIdIndex = electionCandidateIdList.indexOf(candidateId);
+    boolean candidateIncumbency = electionIsIncumbentList.get(candidateIdIndex);
+    String candidatePosition = electionPositionList.get(candidateIdIndex);
+
+    return new Candidate(
+        candidateId,
+        (String) candidateData.getProperty("name"),
+        (String) candidateData.getProperty("partyAffiliation"),
+        (boolean) candidateIncumbency,
```

Comment:
nit: `candidateIncumbency` is already declared boolean on line 92. It doesn't need casting